src/main/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/sentencing/SentencingReconciliationService.kt

```diff
@@ -31,35 +31,52 @@ class SentencingReconciliationService(
     val remandAdjustmentTypes = listOf("RSR", "RX")
   }
 
-  suspend fun generateReconciliationReport(allPrisoners: Boolean, activePrisonersCount: Long): List<MismatchSentencingAdjustments> = activePrisonersCount.asPages(pageSize).flatMap { page ->
-    val activePrisoners = getActivePrisonersForPage(allPrisoners, page)
+  suspend fun generateReconciliationReport(allPrisoners: Boolean): List<MismatchSentencingAdjustments> {
+    val mismatches: MutableList<MismatchSentencingAdjustments> = mutableListOf()
+    var lastBookingId = 0L
 
-    withContext(Dispatchers.Unconfined) {
-      activePrisoners.map { async { checkBookingAdjustmentsMatch(it) } }
+    do {
```

Comment:
replaced bu do..while - keeping calling page end point until we get to the last page